engine/src/main/java/org/terasology/internet/InternetManager.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.internet;
+
+import java.io.IOException;
+
+/**
+ * @author soniex2
+ */
+public interface InternetManager {
+
+    /**
+     * Opens a TCP connection.
+     *
+     * @param hostname The target hostname.
+     * @param port     The target port.
+     * @return The socket.
+     * @throws IOException If an I/O error occurs.
+     */
```

Comment:
I would prefer to return the actual socket and leave the in/out-handling to the caller.